CRM/Case/ActionMapping.php

```diff
@@ -126,7 +137,40 @@ public function createQuery($schedule, $phase, $defaultParams) {
       throw new CRM_Core_Exception("Invalid date field");
     }
 
-    $query['casDateField'] = $schedule->start_action_date;
+    if ($schedule->start_action_date == 'case_status') {
+      // For this case, we use activity of type 'Change Case Status' and check if the case status has been changed
+      // from an indifferent status to the one configured in scheduled reminder.
+      $query->join('cac', "INNER JOIN civicrm_case_activity cac ON cac.case_id = c.id");
+      $query->where("cac.id = (SELECT MAX(cac2.id) FROM civicrm_case_activity cac2 WHERE cac2.case_id = c.id)");
+
+      $query->join('act', "INNER JOIN civicrm_activity act ON act.id = cac.activity_id");
+      $query->join('opt', "INNER JOIN civicrm_option_value opt ON opt.value = act.activity_type_id");
+      $query->join('opg', "INNER JOIN civicrm_option_group opg ON opg.id = opt.option_group_id");
+      $query->where("opt.name = 'Change Case Status'");
+      $query->where("opg.name = 'activity_type'");
+
+      // If start condition is 'after' we check for completed activity
+      // and if 'before' we check for scheduled activity.
+      if ($schedule->start_action_condition == 'after') {
+        $activityStatus = 'Completed';
+      }
+      else {
+        $activityStatus = 'Scheduled';
+      }
+
+      // Subquery to check for activity status.
+      $subQuery = \CRM_Utils_SQL_Select::from("civicrm_option_value opt2")->select('opt2.value');
+      $subQuery->join('opg2', 'INNER JOIN civicrm_option_group opg2 ON opg2.id = opt2.option_group_id');
+      $subQuery->where("opt2.name = @activityStatus")->param('activityStatus', $activityStatus);
+      $subQuery->where("opg2.name = 'activity_status'");
+      $query->where("act.status_id = (" . $subQuery->toSQL() . ")");
```

Comment:
Can't we build the activity statuses from this `CRM_Activity_BAO_Activity::buildOptions('status_id')` and use that in the query for the `IN` rather than using a subquery

Reply:
I tried using that but I cant really get the query result that I am looking. Maybe you can explain a bit more on that subquery part..